notebooks/notebook_template_review.py

```diff
@@ -478,443 +534,339 @@ def validate(self, path: str, cells: list) -> None:
+            'Vertex Dataset': 'Vertex AI Dataset',
+            'Vertex Model': 'Vertex AI Model',
+            'Vertex Endpoint': 'Vertex AI Endpoint',
+            'Vertex Private Endpoint': 'Vertex AI Private Endpoint',
+            'Automl': 'AutoML',
+            'AutoML Tables': 'AutoML Tabular',
+            'AutoML Vision': 'AutoML Image',
+            'AutoML Language': 'AutoML Text',
+            'Tensorflow': 'TensorFlow',
+            'Tensorboard': 'TensorBoard',
+            'Google Cloud Notebooks': 'Vertex AI Workbench Notebooks',
+            'BQ ': 'BigQuery',
+            'BQ.': 'BigQuery',
+            'Bigquery': 'BigQuery',
+            'BQML': 'BigQuery ML',
+            'GCS ': 'Cloud Storage',
+            'GCS.': 'Cloud Storage',
+            'Google Cloud Storage': 'Cloud Storage',
+            'Pytorch': 'PyTorch',
+            'Sklearn': 'scikit-learn',
+            'sklearn': 'scikit-learn'
+        }
+        cell = notebook.get()
         
-    for word in words[1:]:
-        word = word.replace(':', '').replace('(', '').replace(')', '')
-        if word in ACRONYMS:
-            continue
-        if word.isupper():
-            report_error(path, ErrorCode.ERROR_HEADING_CASE, f"heading is not sentence case: {word}")
+        for line in cell['source']:
+            # HTML code
+            if '<a ' in line:
+                continue
 
+            if 'TODO' in line or 'WIP' in line:
+                notebook.report_error(ErrorCode.ERROR_TWRULE_TODO, f'TODO in cell: {line}')
+            if 'we ' in line.lower() or "let's" in line.lower() in line.lower():
+                notebook.report_error(ErrorCode.ERROR_TWRULE_FIRSTPERSON, f'Do not use first person (e.g., we), replace with 2nd person (you): {line}')
+            if 'will' in line.lower() or 'would' in line.lower():
+                notebook.report_error(ErrorCode.ERROR_TWRULE_FUTURETENSE, f'Do not use future tense (e.g., will), replace with present tense: {line}')
 
-def get_cell(path: str, 
-             cells: list, 
-             cell_index: int) -> (list, int):
-    """
-        Get the next notebook cell.
-        
-        path: used only for reporting an error
-        cells: The content cells (JSON) for the notebook
-        cell_index: The index of the last cell that was parsed (reviewed).
-        
-        Returns:
-        
-        cell: content of the next cell
-        cell_index + 1: index of subsequent cell
-    """
-    while empty_cell(path, cells, cell_index):
-        cell_index += 1
-        
-    cell = cells[cell_index]
-    if cell['cell_type'] == 'markdown':
-        check_text_cell(path, cell)
-    return cell, cell_index + 1
+            for mistake, brand in branding.items():
+                if mistake in line:
+                    notebook.report_error(ErrorCode.ERROR_TWRULE_BRANDING, f"Branding {mistake} -> {brand}: {line}")
 
+class SentenceCaseTWRule(NotebookRule):
+    def validate(self,
```

Comment:
The `validate` method must remain the same signature as defined in `NotebookRule` or else it's meaningless.

Reply:
done